src/main/java/edu/harvard/iq/dataverse/MailServiceBean.java

```diff
@@ -164,16 +168,24 @@ public void sendMail(String from, String to, String subject, String messageText)
         sendMail(from, to, subject, messageText, new HashMap<>());
     }
 
-    public void sendMail(String from, String to, String subject, String messageText, Map<Object, Object> extraHeaders) {
+    public void sendMail(String reply, String to, String subject, String messageText, Map<Object, Object> extraHeaders) {
         try {
             MimeMessage msg = new MimeMessage(session);
-            if (from.matches(EMAIL_PATTERN)) {
-                msg.setFrom(new InternetAddress(from));
+            //Always send from system address to avoid email being blocked
+            InternetAddress fromAddress=getSystemAddress();
+            try {
+              fromAddress.setPersonal(fromAddress.getPersonal() + "on behalf of " + reply, charset);
+            } catch (UnsupportedEncodingException ex) {
+                logger.severe(ex.getMessage());
+            }
+            msg.setFrom(fromAddress);
+             
+            if (reply.matches(EMAIL_PATTERN)) {
+            	//But set the reply-to address to direct replies to the requested 'from' party if it is a valid email address	
+                msg.setReplyTo(new Address[] {new InternetAddress(reply)});
             } else {
-                // set fake from address; instead, add it as part of the message
-                //msg.setFrom(new InternetAddress("invalid.email.address@mailinator.com"));
-                msg.setFrom(getSystemAddress());
-                messageText = "From: " + from + "\n\n" + messageText;
+                //Otherwise include the invalid 'from' address in the message
+                messageText = "From: " + reply + "\n\n" + messageText;
```

Comment:
@mheppler you want this "From:" line to go away, right?

Reply:
@pdurbin Maybe someday, but please it as is for this issue. I will open a new issue to track if there is a problem there. Thanks.

Reply:
@mheppler ok, I see you opened #4601. Thanks.